client/src/controllers/MessagesController.ts

```diff
@@ -0,0 +1,46 @@
+import { Controller } from '@hotwired/stimulus';
+
+export class MessagesController extends Controller<HTMLFormElement> {
+  static classes = ['newMessage', 'show'];
+  static targets = ['messages', 'templates'];
+
+  declare readonly hasMessagesTarget: boolean;
+  declare readonly hasNewMessageClass: boolean;
+  declare readonly hasShowClass: boolean;
+  declare readonly hasTemplatesTarget: boolean;
+  declare readonly messagesTarget: HTMLElement;
+  declare readonly newMessageClass: string;
+  declare readonly showClass: string;
+  declare readonly templatesTarget: HTMLElement;
+
+  add(event) {
+    const status = event.detail.status;
+    const statusText = event.detail.text;
```

Comment:
```suggestion
  add(
    event?: CustomEvent<{
      /** Message status level, based on Django's message types. */
      status?: 'success' | 'error' | 'warning' | string;
      /** Content for the message, HTML not supported. */
      text?: string;
    }>,
  ) {
    const { status = 'success', text: statusText } = event?.detail || {};
    if (!statusText) return;
```

Let's use some destructuring and TypeScript here to make this both simpler, more robust and more helpful when knowing that the variables are. We also want to make sure that if the event is not supplied, there is some default behaviour.

Reply:
although when assigning it i do need to use `spanElement.textContent = statusText as string`